libs/base/utilities/src/decorators/hydratable/hydratable.ts

```diff
@@ -63,24 +65,47 @@ function hydratableClass<T extends Type<HTMLElement>>(
     private hasSsr?: boolean;
     private [HYDRATION_CALLS] = 0;
 
+    private [SIGNAL_META]!: {
+      effectRuns: number;
+      renderRuns: number;
+      effect?: Effect;
+    };
+
     static properties = {
       useRealRender: { type: Boolean, state: true },
     };
 
     constructor(...args: any[]) {
       super(...args);
+      new SignalController(this as any);
+
       this.hasSsr = !isServer && !!this.shadowRoot;
       this.useRealRender = true;
 
       if (isServer) {
         this.setAttribute(hydratableAttribute, mode ?? '');
+
+        this[SIGNAL_META] = {
+          effectRuns: 0,
+          renderRuns: 0,
+        };
       }
 
       if (this.hasSsr) {
         this[DEFER_HYDRATION] = true;
       }
     }
 
+    willUpdate(_changedProperties: PropertyValues): void {
+      super.willUpdate(_changedProperties);
+      if (isServer) {
+        this[SIGNAL_META].effect = effect(() => {
+          this[SIGNAL_META].effectRuns++;
+          this.render();
```

Comment:
Is it mean that we have two renders on ssr, when we call it in `willUpdate` and when lit call it?

Reply:
we have only one render on SSR, but we can run render method multiple times on a server, if signal used in a render will trigger async value (and change of any signal used in last render run)